juspay/hyperswitch
# feat(connector): accept connector_transaction_id in error_response of connector flows for Trustpay

## Type of Change



## Description

Resolves [#3543](https://github.com/juspay/hyperswitch-cloud/issues/3543)


## How did you test it?

1. Create a failed payment with cards in trustpay
    Test card number `4200 0000 0000 0002`
    for further test cards, refer [here]( https://doc.trustpay.eu/#cards-test-cards)
    Expected behavior :  `connector_transaction_id` should be populated

<img width="598" alt="Screenshot 2023-12-05 at 8 04 40 PM" src="https://github.com/juspay/hyperswitch/assets/131388445/5f09e4ce-7685-440b-b475-a1fb6ed281e6">

<img width="467" alt="Screenshot 2023-12-05 at 8 04 48 PM" src="https://github.com/juspay/hyperswitch/assets/131388445/2550344f-b7b6-4b8d-9f10-f0f70a398359">


3. Create a bank redirect payment with trustpay and fail the payment in redirect. Expected behavior:  `connector_transaction_id` should be populated (Note: For some types error, connector transaction id will not be sent by the connector)

<img width="805" alt="Screenshot 2023-12-05 at 8 05 06 PM" src="https://github.com/juspay/hyperswitch/assets/131388445/91848f97-5ff3-40ab-ba76-406f6f7a3bf0">


## Checklist
